core/network/firewall/rule.go

```diff
@@ -251,3 +251,14 @@ func (rules IngressRules) RemoveCIDRsMatchingAddressType(removeAddrType network.
 	uniqueRules.Sort()
 	return uniqueRules
 }
+
+// EnsureSourceCIDRs fills in a default IPv4 CIDR ("0.0.0.0/0") for any rule
+// that has an empty SourceCIDRs slice. This ensures firewall rules always have
+// at least one source range.
+func (rules IngressRules) EnsureSourceCIDRs() {
```

Comment:
This is a bit misleading as it only deals with ipv4. It's only being used as a helper in the gce provider so it doesn't pay the cost to expose it here and have to test it separately etc.

Reply:
yeah, sort of.
Let me move this into internal/provider/gce/environ_firewall.go then